project_task_materials_stock/models/project.py

```diff
@@ -0,0 +1,169 @@
+                self.material_ids.create_analytic_line()
+            else:
+                self.unlink_stock_move()
+                self.analytic_line_ids.unlink()
+        return res
+
+    @api.multi
+    def action_assign(self):
+        self.stock_move_ids.action_assign()
+
+    @api.multi
+    def action_done(self):
+        self.stock_move_ids.action_done()
+        pass
+
+
+class ProjectTaskMaterials(models.Model):
+    _inherit = "project.task.materials"
+
+    stock_move_id = fields.Many2one(
+        comodel_name='stock.move', string='Stock Move')
+    analytic_line_id = fields.Many2one(
+        comodel_name='account.analytic.line', string='Analytic Line')
+    product_uom = fields.Many2one(
+        comodel_name='product.uom', string='Unit of Measure')
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.product_uom = self.product_id.uom_id
+        return {'domain': {'product_uom': [
+            ('category_id', '=', self.product_id.uom_id.category_id.id)]}}
+
+    def uos_qty(self):
+        if self.product_uom and self.product_id.uom_id and \
+                (self.product_uom != self.product_id.uom_id):
+            res = self.quantity * self.product_id.uom_id.factor
+            res = res / self.product_uom.factor
+            res = float_round(
+                res, precision_rounding=self.product_uom.rounding,
+                rounding_method='UP')
+        else:
+            res = self.quantity
+        return res
+
+    def _prepare_stock_move(self):
+        product = self.product_id
+        res = {
+            'product_id': product.id,
+            'name': product.partner_ref,
+            'state': 'confirmed',
+            'product_uom': self.product_uom.id or product.uom_id.id,
+            'product_uos': self.product_uom.id,
+            'product_uom_qty': self.quantity,
+            'product_uos_qty': self.quantity,
+            'origin': self.task_id.name,
+            'location_id':
+            product.product_tmpl_id.property_stock_procurement.id,
+            'location_dest_id': self.env.ref(
+                'stock.stock_location_customers').id,
+        }
+        return res
+
+    @api.one
+    def create_stock_move(self):
+        move_id = self.env['stock.move'].create(self._prepare_stock_move())
+        self.stock_move_id = move_id.id
+
+    def _prepare_analytic_line(self):
+        product = self.product_id
+        company_id = self.env['res.company']._company_default_get(
+            'account.analytic.line')
+        journal = self.env.ref(
+            'project_task_materials_stock.analytic_journal_sale_materials')
+        res = {
+            'name': self.task_id.name + ': ' + product.name,
+            'ref': self.task_id.name,
+            'product_id': product.id,
+            'journal_id': journal.id,
+            'unit_amount': self.quantity,
+            'account_id': self.task_id.project_id.analytic_account_id.id,
```

Comment:
I was thinking about it, and I guess that the Timesheets integrated with Tasks surely have the same problem.
Given this, I'm wondering if the design approach for the Service Desk feature is the best one.
Probably this deserves a discussion on it's own...